exporters/aws-sq-exporter/sq_exporter.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/python3
+
+import subprocess, os
+import json
+from collections import namedtuple
+from prometheus_client import start_http_server, Summary, Gauge
+import argparse
+import time
+import boto3
+import bios
+from benedict import benedict
+
+#Generic function to fetch administrative quota values
+def getQuotaValue(quotaCode,serviceCode,cSessions):
+   paginator = cSessions["service-quotas"].get_paginator('list_service_quotas')
+   pCursor = paginator.paginate(ServiceCode=serviceCode, PaginationConfig={'MaxItems': 1000 , 'PageSize': 10})
+   currentValue=0
+   currentQ = 0
+   for page in pCursor:
+      for quotas in page['Quotas']:
+         if quotas['QuotaCode'] == quotaCode:
+             currentQ = str(quotas['Value'])
+         currentValue= currentValue+1
+   return currentQ
+
+# fetch actual usage of specific service, works for EIP and Vpcs
+def getUsage(cSessions,usageRetrieval,usageFilter):
+   awsCall = getattr(cSessions["ec2"],usageRetrieval)
+   awsReturns = awsCall()
+   return len(awsReturns[usageFilter])
+
+def getAccountID():
+  awsSession = boto3.client("sts", aws_access_key_id=args.apikey, aws_secret_access_key= args.secretkey)
+  awsReturns = awsSession.get_caller_identity()
+  return awsReturns["Account"]
+
+## If we want to fetch the usage for all of the regions on given account
+## we'll need to fetch a list of regions available on this particular AWS account
+def getRegions():
+   awsSession = boto3.client("ec2", aws_access_key_id=args.apikey, aws_secret_access_key= args.secretkey)
+   awsReturns = awsSession.describe_regions()
+   if args.debug == True:
+       print("Regions fetched from active account: "+str(awsReturns))
+   regions = []
+   for page in awsReturns["Regions"]:
+       regions.append(page["RegionName"])
+       if args.debug == True:
+           print("Adding "+str(page["RegionName"])+" to the region list")
+   return regions
+
+
+if __name__ == '__main__':
+   # Fetch&parse args
+   parser = argparse.ArgumentParser()
+   parser.add_argument("-a","--apikey",help=" AWS Access Key ID ")
+   parser.add_argument("-s","--secretkey", help=" AWS Sercet Access Key")
+   parser.add_argument("-r","--region", default="none", help=" AWS Region to be used fo queries")
+   parser.add_argument("-t","--time", type=int, default=5 , help=" Sleep time between fetching the AWS API input")
```

Comment:
I'd say we should default this to 900 (15 minutes) as the consumption aren't expected to change that quickly

Reply:
Applied